src/raster/gcp.rs

```diff
@@ -0,0 +1,37 @@
+//! Raster ground control point support
+
+use crate::spatial_ref::SpatialRef;
+use crate::Dataset;
+
+impl Dataset {
+    /// Get output spatial reference system for GCPs.
+    ///
+    /// # Notes
+    /// * This is separate and distinct from [`Dataset::spatial_ref`], and only applies to
+    /// the representation of ground control points, when embedded.
+    ///
+    /// See: [`GDALGetGCPSpatialRef`](https://gdal.org/api/raster_c_api.html#_CPPv420GDALGetGCPSpatialRef12GDALDatasetH)
+    pub fn gcp_spatial_ref(&self) -> Option<SpatialRef> {
+        let c_ptr = unsafe { gdal_sys::GDALGetGCPSpatialRef(self.c_dataset()) };
+
+        if c_ptr.is_null() {
+            return None;
+        }
+
+        unsafe { SpatialRef::from_c_obj(c_ptr) }.ok()
```

Comment:
I wonder if this is leaking the `OGRSpatialReferenceH` (check out the implementation of `from_c_obj`).

Does `foreign_types` really solve issues like this? I only skimmed your changes, but it would be great if it did.

Reply:
That would be awesome, I'll try to look into it.